### ISSUE TITLE
Players go offline.

### ISSUE THREAD

**Issue by yvasquezmonoku** [oculus-samples/Unreal-SharedSpaces#29]
My players get disconnected when leaving the Guardian area. Is there anything that can be done to prevent this from happening?

**Comment by els-oculus** [comment]
That's what you want to happen.   When you leave the guardian area, the app is suspended, so it is the same behavior as if you remove your headset.   If the person that leaves the guardian area happens to be the host, then hosting is transferred to another player.

This said, I see that respawning doesn't put you back where you were, that can definitely be improved.

**Comment by yvasquezmonoku** [comment]
okay, I got it. I thought it was some problem. One bonus question. Is it normal that some guest players take time to be teleported to the level where the host entered?
In my case when they are in the lobby and the host goes to a new room, he pulls everyone who is in the lobby to this room.

**Comment by els-oculus** [comment]
That's not how SharedSpaces is built.   When the hosting player goes to a different room, he joins-or-creates the new destination and the players are left in their current room, with a new player hosting (i.e. the new Photon master client).

What you describe here gives me the impression that your hosting player simply loads a new map without going through a full reconnection flow.   When the host loads a new map, that map will be loaded on all clients, too.   But again, that's not how the original SharedSpaces sample works, that's particular to your implementation.

**Comment by yvasquezmonoku** [comment]
okay. Do you mean that the operation of shared spaces is that if the host goes from the lobby to another room, the guests will not carry him, but they must enter the room by themselves?

**Comment by els-oculus** [comment]
Correct: when player X (who is hosting a Lobby or whatever other room) leaves, the remaining players stay where they are.   They can follow player X to his new destination, of course.   Watch the video we made, you will see that in action.
